.github/workflows/pr.yml

```diff
@@ -177,12 +183,64 @@ jobs:
       - name: Untar artifacts
         shell: bash
         run:  tar xfz target.tgz
-      - name: setup node
-        shell: bash
-        run: (cd ui/client; npm ci)
       - name: Prepare docker
         shell: bash
         run: echo "version in ThisBuild := \"$GITHUB_SHA\"" > version.sbt && ./ciRunSbt.sh dist/docker:publishLocal
       - name: Test docker
         shell: bash
-        run: cd ./demo/docker && echo "NUSSKNACKER_VERSION=$GITHUB_SHA" > .env && ./testQuickstart.sh
+        run: cd ./demo/docker && echo "NUSSKNACKER_VERSION=ghactionstest-$GITHUB_SHA" > .env && ./testQuickstart.sh
+
+#TODO: extract to different workflow?
+  publish:
+    runs-on: ubuntu-latest
+    needs: [build, tests, crossCompile, integrationTests, slowTests, frontendTests, dockerTest]
+    if: ${{ github.ref == 'refs/heads/staging' || github.ref == 'refs/heads/master' || startsWith(github.head_ref, 'preview') ||  github.head_ref == 'publish_githubactions' }}
+    steps:
+      - name: debug
+        env:
+          GITHUB_CONTEXT: ${{ toJson(github) }}
+        run: echo "$GITHUB_CONTEXT"
+
+      - uses: actions/checkout@v1
+      - name: Cache npm
+        uses: actions/cache@v2
+        with:
+          path: |
+            **/node_modules
+          key: ${{ runner.os }}-${{ hashFiles('**/package-lock.json') }}
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: 14.11.0
+      - name: Setup npm
+        run: (cd ui/client; npm ci)
+      - name: Cache ivy packages
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.ivy2/cache
+            ~/.sbt
+          key: ${{ runner.os }}-ivy2-${{ hashFiles('**/*.sbt') }}
+          restore-keys: ${{ runner.os }}-sbt
+      - uses: olafurpg/setup-scala@v10
+      - uses: actions/download-artifact@v2
+        with:
+          name: build-target
+      - name: Untar artifacts
+        shell: bash
+        run:  tar xfz target.tgz
+      - name: Login to Docker Hub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USER }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Build
+        shell: bash
+        env:
+          #TODO: shouldn't be needed...
+          nexusUrl: https://oss.sonatype.org/content/repositories/snapshots
+          nexusUser: ${{ secrets.SONATYPE_USER }}
+          nexusPassword: ${{ secrets.SONATYPE_PASSWORD }}
+        #TODO: handle version better
+        run: sbt -J-Xms1000M -J-Xmx1000M "set version in ThisBuild := (version in ThisBuild).value.replace(\"-SNAPSHOT\", \"-$(date -I)-$GITHUB_RUN_NUMBER-$GITHUB_SHA-ghactionstest-SNAPSHOT\")" publish dist/docker:publish
```

Comment:
Yeap, it's ugly.. But for now we can leave it.

Reply:
:)